.github/ISSUE_TEMPLATE/feature_proposal.md

```diff
@@ -0,0 +1,39 @@
+---
+name: Feature proposal
+about: Propose a great new feature for Mitiq
+title: ''
+labels: 'enhancement'
+assignees: ''
+
+---
+## Pre-Request Checklist
+
+- [ ] I checked to make sure that this feature has not already been requested
+- [ ] I or my organization would be willing to help implement and/or test this.
+- [ ] This is **not** a breaking change to Mitiq.
+
+## Feature description
+
+_TODO: Insert a short outline for what the proposed modification is._
+
+## Considerations
```

Comment:
Could we rename this section to "Purpose(s) of feature" or something similar? "Considerations"  is a bit vague to me.

Reply:
My intent is that it is broader than just purpose, in that it also asks about pros and cons of the approach outlined in the previous section and if there are other possible approaches to the problem the proposal. The purpose should have really be already described in the description of the feature. Maybe `##Discussion`` would be a better heading?